core/src/main/scala/org/apache/spark/Accumulators.scala

```diff
@@ -314,6 +409,59 @@ object AccumulatorParam {
   // TODO: Add AccumulatorParams for other types, e.g. lists and strings
 }
 
+/**
+ * A consistent wrapper of [[org.apache.spark.AccumulatorParam]] where we keep track of
+ * RDD/partitions which have already been processed
+ *
+ * @tparam T type of value to accumulate
+ */
+class ConsistentAccumulatorParam[T](accumulatorParam: AccumulatorParam[T])
```

Comment:
I think you should create test cases for this class, and don't depend on the framework to handle some unhandled cases